p2p/rlpx.go

```diff
@@ -694,33 +694,16 @@ func (rw *rlpxFrameRW) ReadMsg() (msg Msg, err error) {
 		if err != nil {
 			return msg, err
 		}
-		msg.Size = uint32(size)
-		msg.Payload = &snappyReader{payload: bytes.NewReader(payload)}
-	}
-	return msg, nil
-}
-
-// snappyReader is a lazy decompressor that expands a snappy encoded stream
-// only upon first read request. This is useful to allow protocols to skip
-// large messages without having to decode them (DOS protection).
-type snappyReader struct {
-	payload io.Reader
-	decoded bool
-}
-
-func (r *snappyReader) Read(p []byte) (n int, err error) {
-	if !r.decoded {
-		payload, err := ioutil.ReadAll(r.payload)
-		if err != nil {
-			return 0, err
+		if size > int(maxUint24) {
+			return msg, errors.New("decompressed message size overflows uint24")
```

Comment:
Heh, "decompressed message exceeds 16MB" is too user-friendly ?

Reply:
:)) The user won't see this. But the existing counterpart in write is "message size exceeds uint24".

Though you made me realize I need a check in write too before I compress.